### PR TITLE
[release/1.x] [table] measureTextContentWithExclusions

### PR DESCRIPTION
**Pull request by adidahiya**
### Motivation

We'd like to add some text to our column headers (column descriptions) which should be shown in the header cell (as an affordance to users that they can click and view more details about the column) but not taken into account for double-click-to-resize interactions on the column.

### CODE REVIEW

File: packages/table/src/common/utils.ts [hunk 1]
```diff
@@ -408,3 +407,27 @@ export const Utils = {
...
+ * Wrapper around document.getComputedStyle which applies some extra logic to exclude certain elements
+ * from the computation.
+ */
+function getComputedStyleWithExclusions(element: HTMLElement) {
+    const elementsToExclude = element.querySelectorAll(`.${CLASSNAME_EXCLUDED_FROM_TEXT_MEASUREMENT}`);
+
+    // hide all elements to exclude
+    for (let i = 0; i < elementsToExclude.length; i++) {
+        const el = elementsToExclude.item(i) as HTMLElement;
+        el.setAttribute("style", "display: none");
```
Comment (adidahiya) [inline packages/table/src/common/utils.ts]: we should save off the old `style` attribute value and set it back to what it was previously.

**Comment by adidahiya** [comment]
h/o, this won't work, give me a min

File: packages/table/src/common/utils.ts [hunk 2]
```diff
@@ -408,3 +407,27 @@ export const Utils = {
...
+
+/**
+ * Wrapper around document.getComputedStyle which applies some extra logic to exclude certain elements
+ * from the computation.
+ */
+function getComputedStyleWithExclusions(element: HTMLElement) {
+    const elementsToExclude = element.querySelectorAll(`.${CLASSNAME_EXCLUDED_FROM_TEXT_MEASUREMENT}`);
+
+    // hide all elements to exclude
+    for (let i = 0; i < elementsToExclude.length; i++) {
```
Comment (cmslewis) [inline packages/table/src/common/utils.ts]: nit: use `forEach`?

**Comment by adidahiya** [comment]
updated to do the "ignore me" part in the right spot

File: packages/table/src/common/utils.ts [hunk 3]
```diff
@@ -184,17 +186,14 @@ export const Utils = {
...
      * We use the computed font from the supplied element and a non-DOM canvas
      * context to measure the text.
-     *
-     * Returns a `TextMetrics` object.
      */
-    measureElementTextContent(element: Element) {
+    measureElementTextContent(element: Element): TextMetrics {
         const context = document.createElement("canvas").getContext("2d");
-        const style = getComputedStyle(element, null);
+        const style = getComputedStyle(element);
```
Comment (adidahiya) [inline packages/table/src/common/utils.ts]: should probably add back the second param

File: packages/table/src/locator.ts [hunk 4]
```diff
@@ -101,7 +101,9 @@ export class Locator implements ILocator {
 
     public getWidestVisibleCellInColumn(columnIndex: number): number {
         const columnCellSelector = this.getColumnCellSelector(columnIndex);
-        const columnHeaderAndBodyCells = this.tableElement.querySelectorAll(columnCellSelector);
+        const columnHeaderAndBodyCells = this.tableElement.querySelectorAll(columnCellSelector) as NodeListOf<
+            HTMLElement
+        >;
```
Comment (adidahiya) [inline packages/table/src/locator.ts]: this change is no longer necessary

**Comment by adidahiya** [comment]
@giladgray or @themadcreator can I get a quick +1? @tgreenwatts has already tested this and it works

File: packages/table/src/common/utils.ts [hunk 5]
```diff
@@ -42,7 +44,7 @@ export const Utils = {
      * element's original className and any other classes passed in with variadic
      * arguments matching the `classNames` api.
      */
-    assignClasses<P extends IProps>(elem: React.ReactElement<P>, ...extendedClasses: ClassValue[]) {
+    assignClasses<P extends IProps>(elem: React.ReactElement<P>, ...extendedClasses: ClassValue[]): React.ReactNode {
```
Comment (rcchen) [inline packages/table/src/common/utils.ts]: Why all the explicit return signatures?

Comment (adidahiya) [reply packages/table/src/common/utils.ts]: I find it good hygiene for public-ish APIs. Not entirely relevant to the change at hand.

**Pull request merged**